app/models/miq_worker/runner.rb

```diff
@@ -332,7 +332,10 @@ def do_work_loop
       begin
         heartbeat
         do_work
-      rescue TemporaryFailure
+      rescue TemporaryFailure => error
+        msg = "#{log_prefix} Temporary failure (message: '#{error}') caught"\
+            " during #do_work. Sleeping for a while before resuming."
+        _log.warn(msg)
```

Comment:
> Should this follow a similar pattern to authentication check at boot where we won't start if the creds are bad or if they system is temporarily not available?

I'd consider a temporary failure no reason to exit, so keep it like this. 
And `TopicNotFound` and `ProviderUnreachable` can be considered as Temporary.